### ISSUE TITLE
Time Machine volume size limit

### ISSUE THREAD

**Issue by RlndVt** [rockstor/rockstor-core#2343]
I would love to be able to set a volume size limit for a Time Machine share, so it doesn't use the entire array.

To quote from [the docker-timemachine github:](https://github.com/mbentley/docker-timemachine#optional-variables-for-smb) 

> [VOLUME_SIZE_LIMIT] sets the maximum size of the time machine backup; a unit can also be passed (e.g. - 1 T). See the [Samba Docs](https://www.samba.org/samba/docs/current/man-html/vfs_fruit.8.html)  under the fruit:time machine max size section for more details

Thank you for your consideration. 

PS, is it possible to manually edit samba share config files?

**Comment by RlndVt** [comment]
To answer my PS question, according to [the documentation](https://rockstor.com/docs/interface/storage/file_sharing/samba_ops.html), one should be able to using the `Custom configuration` field and adding:

`fruit:time machine max size = SIZE [K|M|G|T|P]`
e.g.
`fruit:time machine max size = 500 G`

A nice UI component to implement this would be nice.

**Comment by FroggyFlox** [comment]
Hi @RlndVt,
I was going to point you towards the "Custom configuration" field but it appears I was too slow at it ;)

> A nice UI component to implement this would be nice.

Thanks for your feedback here, actually... We always strive to keep the UI as user friendly as possible yet allowing as much customization as possible. The custom configuration field does just that but I agree that it is not the most user friendly as one needs to know before hand what to enter there.
This is where feedback like yours is particularly valuable as it tells us what options really matter to the users and are thus worth offering a simple UI element for it. I was actually pondering the idea of improving our macOS compatibility from the webUI lately following some user feedback so that could definitely be part of it. For these reasons, thank you very much for taking the time to create this issue, this is very valuable to the project!

Following-up on this, do not hesitate to share your experience and any additional feedback you may have on the forum. Rockstor is community-driven project so development will go where user feedback is. For instance, if there are any additional Samba options you routinely find yourself adding, maybe it's worth adding a UI element for them. This is especially true for our users accessing samba exports from macOS clients as this is something more difficult to test for us at the moment.

Bottom line, thanks a lot for your help here!
